onnxruntime/contrib_ops/cpu/bert/fast_gelu.cc

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "fast_gelu.h"
+
+#include "core/common/safeint.h"
+#include "core/framework/tensor.h"
+#include "core/platform/threadpool.h"
+#include "core/providers/common.h"
+#include "core/util/math_cpuonly.h"
+#include "core/mlas/inc/mlas.h"
+#include "fast_gelu_helper.h"
+
+namespace onnxruntime {
+namespace contrib {
+
+ONNX_OPERATOR_KERNEL_EX(
+    FastGelu,
+    kMSDomain,
+    1,
+    kCpuExecutionProvider,
+    KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<float>()),
+    FastGelu<float>);
+
+// FastGelu = 0.5 * (1 + Tanh(x * (C * x * x + B))) * X
+static constexpr float B = 0.7978845608028654f;    // sqrt(2.0 / M_PI)
+static constexpr float C = 0.035677408136300125f;  // 0.044715 * sqrt(2.0 / M_PI)
+
+template <typename T>
+Status FastGelu<T>::ComputeGelu(OpKernelContext* context, const T* input_data, T* output_data, int64_t elem_count) const {
```

Comment:
the difference with BiasGelu is the function to compute Gelu. It would be great to have a Base class for BiasGelu and FastGelu, and let BiasGelu and FastGelu override the function.

Reply:
I merged them into one class to avoid duplicated code, and there is benefit that it is easy to merge them into one operator in the future (if needed).